### PR TITLE
Bugfix: Make houdini submissions respect pools groups

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
Make houdini submissions respect pools groups.
This is done by:
1. Make collect pools works with some Houdini families/product types.
2. Make Ayon Houdini submitters get group names from Houdini deadline settings.

## Additional info
I made three Houdini deadline setting categories. each for a particular case.
image

## Testing notes:
1. Set pools and groups in ayon deadline addon settings
3. Publish render and cache to deadline.

### CODE REVIEW

File: server_addon/deadline/server/settings/publish_plugins.py [hunk 1]
```diff
@@ -211,6 +211,49 @@ class HarmonySubmitDeadlineModel(BaseSettingsModel):
...
+    chunk_size: int = SettingsField(title="Chunk Size")
+    group: str = SettingsField(title="Group")
+
+
+class HoudiniSubmitDeadlineModel(BaseSettingsModel):
+    """Houdini deadline submitter settings."""
+    enabled: bool = SettingsField(title="Enabled")
+    optional: bool = SettingsField(title="Optional")
+    active: bool = SettingsField(title="Active")
+    HoudiniSubmitCacheDeadline: HoudiniSubmitCacheDeadlineModel = (
```
Comment (iLLiCiTiT) [inline server_addon/deadline/server/settings/publish_plugins.py:237]: Why do you nest them under single model? You then have to do this https://github.com/ynput/ayon-core/pull/10/files#diff-e7f176c29a36e99c26e9934dd0a1acfab0b28dc7f5423f7edc85735f871456c0R88

File: server_addon/deadline/server/settings/publish_plugins.py [hunk 2]
```diff
@@ -306,36 +349,41 @@ class PublishPluginsModel(BaseSettingsModel):
...
+            section="Hosts"
         )
     )
-    CelactionSubmitDeadline: CelactionSubmitDeadlineModel = SettingsField(
-        default_factory=CelactionSubmitDeadlineModel,
-        title="Celaction Submit Deadline")
     BlenderSubmitDeadline: BlenderSubmitDeadlineModel = SettingsField(
         default_factory=BlenderSubmitDeadlineModel,
         title="Blender Submit Deadline")
+    CelactionSubmitDeadline: CelactionSubmitDeadlineModel = SettingsField(
```
Comment (iLLiCiTiT) [inline server_addon/deadline/server/settings/publish_plugins.py:362]: I think the order was determined. I don't think the change of the order should not happen in this PR.

File: server_addon/deadline/server/settings/publish_plugins.py (hunk 1 above)
Comment (MustafaJafar) [reply server_addon/deadline/server/settings/publish_plugins.py:237]: Oh, I was concerned about organising the Houdini settings and missed that!

File: client/ayon_core/modules/deadline/plugins/publish/submit_houdini_cache_deadline.py [hunk 3]
```diff
@@ -46,9 +46,17 @@ class HoudiniCacheSubmitDeadline(abstract_submit_deadline.AbstractSubmitDeadline
...
+    group = None
     jobInfo = {}
     pluginInfo = {}
-    group = None
+
+    @classmethod
+    def apply_settings(cls, project_settings, system_settings):
+        settings = project_settings["deadline"]["publish"]["HoudiniSubmitDeadline"]["HoudiniSubmitCacheDeadline"]  # noqa
+        cls.priority = settings.get("priority", cls.priority)
+        cls.ChunkSize = settings.get("chunk_size", cls.ChunkSize)
```
[the remark this answers is not in the document]
Comment (MustafaJafar) [reply client/ayon_core/modules/deadline/plugins/publish/submit_houdini_cache_deadline.py:58]: short story, I found it camel case and left it as it was 😅.

File: server_addon/deadline/server/settings/publish_plugins.py (hunk 2 above)
Comment (MustafaJafar) [reply server_addon/deadline/server/settings/publish_plugins.py:362]: https://github.com/ynput/ayon-core/pull/17

**Comment by MustafaJafar** [comment]
While I don't know what should I do on deadline server to test this so I printed the job info in ayon log and it seems to have the proper info in the job info. 

```
DEBUG: DeadlineJobInfo(Plugin='Houdini'
 Frames='1001-1001x1'
 Name='robo_BigRobo_cfx_v166.hip - mantra_ropSplit [EXPORT IFD]'
 Comment=''
 Department=None
 BatchName='robo_BigRobo_cfx_v166.hip'
 UserName='Mustafa_Taher'
 MachineName='Major-Kalawy'
 Pool='ayon'
 SecondaryPool='none'
 Group='export_machines'
 Priority=50
 ChunkSize=10
 .
 .
 .

DEBUG: DeadlineJobInfo(Plugin='Mantra'
 Frames='1001-1001x1'
 Name='robo_BigRobo_cfx_v166.hip - mantra_ropSplit [RENDER]'
 Comment=''
 Department=None
 BatchName='robo_BigRobo_cfx_v166.hip'
 UserName='Mustafa_Taher'
 MachineName='Major-Kalawy'
 Pool='ayon'
 SecondaryPool='none'
 Group='render_machines'
 Priority=50
 ChunkSize=1
.
.
.

DEBUG: DeadlineJobInfo(Plugin='Houdini'
 Frames='1001-1001x1'
 Name='robo_BigRobo_cfx_v166.hip - pointcacheBgeoFarmTest [PUBLISH]'
 Comment=''
 Department=None
 BatchName='Robo - robo_BigRobo_cfx_v166.hip'
 UserName='Mustafa_Taher'
 MachineName='Major-Kalawy'
 Pool='ayon'
 SecondaryPool='none'
 Group='cache_machines'
 Priority=50
 ChunkSize=999999
.
.
.

```

For reference, This validator works with Houdini 
image

**Review by moonyuet — DISMISSED** [review]
I have tested the code; it works as expected. it is great to add `use_published` (its label name is Use Published Scene) in AYON settings, similar to other hosts(as I notice there is `use_published =True` in the settings). 
Just some opinion: it would be great to move `get_attribute_defs` function to the end of the code for readability.

**Review by moonyuet — APPROVED** [review]
Looks good now. (My bad as I switch to wrong branch.)
image

File: client/ayon_core/modules/deadline/plugins/publish/submit_houdini_render_deadline.py [hunk 4]
```diff
@@ -103,10 +109,14 @@ def get_attribute_defs(cls):
                 minimum=1,
                 maximum=1000
             ),
+            TextDef("group",
+                    default=cls.group,
+                    label="Group Name"
+            ),
```
Comment (iLLiCiTiT) [inline client/ayon_core/modules/deadline/plugins/publish/submit_houdini_render_deadline.py:115]: Keep formatting the same please.
```suggestion
            TextDef(
                "group",
                 default=cls.group,
                 label="Group Name"
            ),
```

File: client/ayon_core/modules/deadline/plugins/publish/submit_houdini_render_deadline.py [hunk 5]
```diff
@@ -117,11 +127,9 @@ def get_attribute_defs(cls):
...
                 maximum=1000
             ),
-            BoolDef(
-                "suspend_publish",
-                default=False,
-                label="Suspend publish"
-            )
+            TextDef("export_group",
+                    default=cls.export_group,
+                    label="Export Group Name"),
```
Comment (iLLiCiTiT) [inline client/ayon_core/modules/deadline/plugins/publish/submit_houdini_render_deadline.py:132]:
```suggestion
            TextDef(
                "export_group",
                default=cls.export_group,
                label="Export Group Name"
            ),
```

File: server_addon/deadline/server/settings/publish_plugins.py [hunk 6]
```diff
@@ -416,6 +448,25 @@ class PublishPluginsModel(BaseSettingsModel):
...
+    "HoudiniSubmitDeadline":{
+        "enabled": True,
+        "optional": False,
+        "active": True,
+        "priority": 50,
+        "chunk_size":1,
+        "group": "",
+        "export_priority": 50,
+        "export_chunk_size":10,
+        "export_group": ""
```
Comment (iLLiCiTiT) [inline server_addon/deadline/server/settings/publish_plugins.py:468]: Formatting
```suggestion
        "chunk_size": 999999,
        "group": ""
    },
    "HoudiniSubmitDeadline":{
        "enabled": True,
        "optional": False,
        "active": True,
        "priority": 50,
        "chunk_size": 1,
        "group": "",
        "export_priority": 50,
        "export_chunk_size": 10,
        "export_group": ""
```

File: server_addon/deadline/server/settings/publish_plugins.py [hunk 7]
```diff
@@ -416,6 +448,25 @@ class PublishPluginsModel(BaseSettingsModel):
         "group": "",
         "department": ""
     },
+    "HoudiniCacheSubmitDeadline": {
+        "enabled": True,
+        "optional": False,
+        "active": True,
+        "priority": 50,
+        "chunk_size":999999,
```
[the remark this answers is not in the document]
Comment (moonyuet) [reply server_addon/deadline/server/settings/publish_plugins.py:456]: When I made this implementation of the farm cache, it is set to 999999 avoid users to set the chunk size when they have relative huge frame range (although there is a setting for this chunk size)

[the remark this answers is not in the document]
Comment (MustafaJafar) [reply server_addon/deadline/server/settings/publish_plugins.py]: Should I change it in plugin too.. set it to use `999999` ?

**Review by antirotor — APPROVED** [review]

**Pull request merged**